lib/dry/matcher/evaluator.rb

```diff
@@ -20,11 +29,18 @@ def respond_to_missing?(name, include_private = false)
       def method_missing(name, *args, &block)
         return super unless @cases.key?(name)
 
+        @handled.add name
         handle_case @cases[name], *args, &block
       end
 
       private
 
+      def ensure_exhaustive_match
+        if (not_handled = @cases.keys - @handled.to_a).any?
```

Comment:
You can also treat all cases as not handled initially and then remove them one by one. This won't give much but you'll be able to simplify this expression to `if @not_handled.any?`